Makefile

```diff
@@ -14,14 +13,20 @@ debug:
 debug-werror:
 	mkdir -p $(BUILD_DIR) && cd $(BUILD_DIR) && cmake -DWERROR:BOOL=ON -DCMAKE_BUILD_TYPE=Debug .. && $(MAKE) $(MAKE_FLAGS)
 
+debug-lib:
+	mkdir -p $(BUILD_DIR) && cd $(BUILD_DIR) && cmake -DCMAKE_BUILD_TYPE=Debug -DMATA_BUILD_EXAMPLES:BOOL=OFF -DBUILD_TESTING:BOOL=OFF .. && $(MAKE) $(MAKE_FLAGS)
+
 release:
 	mkdir -p $(BUILD_DIR) && cd $(BUILD_DIR) && cmake -DCMAKE_BUILD_TYPE=Release .. && $(MAKE) $(MAKE_FLAGS)
 
 release-werror:
 	mkdir -p $(BUILD_DIR) && cd $(BUILD_DIR) && cmake -DWERROR:BOOL=ON -DCMAKE_BUILD_TYPE=Release .. && $(MAKE) $(MAKE_FLAGS)
 
+release-lib:
```

Comment:
I think in our master make we are starting to have quite a lot of targets. We should add some comment to each of them or find some better names, since these names says fucking nothing.

Reply:
I added some comments, please check if they make sense. But for me using a make file in a cmake project is very weird and I would look for a way to get rid of it.

Reply:
I am with you on that. It is convenient for GAs, but in general seems weird. But if it works now, let us keep it as it is in the foreseeable future.

Reply:
How do you build it then? I would never ever remove master Makefile, I consider it as a standard that is used everywhere for everything. Things like `make install`, `make doc`, etc. are understood by everyone everywhere, and I believe almost every sane person (not considering psychopats and likes) starts with any project by simply typing `make`. So please, lets keep master Makefile :)

Reply:
Like this (taken from [Catch2 build system](https://github.com/catchorg/Catch2/blob/devel/tools/scripts/buildAndTest.sh) – a CMake project without a Makefile) inside some shell script:
```sh
# 2. Configure the full test build
cmake -B build/ -S . -DCMAKE_BUILD_TYPE=Debug --preset all-tests

# 3. Run the actual build
cmake --build build/
```
The problem with having both CMake and Makefile is that one does not know whether to create a new target in one or the other. If you want to add a new build functionality, what do you do? Add a target to CMake and not to Makefile? Add a target to Makefile, but not to CMake? Add a target to both? It is harder to maintain.

This is good enough for now. One day, we may revisit this conversation and think about it again.

Reply:
Once you are as old as us, and every character typed sends pain to you joints, you too will appreciate simple `make release` :D

Reply:
Exactly. That is why you have this in a shell script (as in the example above: this is inside a build and test shell script). You type it once, then you only press an up arrow while on the command line to return to the script execution and press enter. The problem with make is, as described above, that you have two build systems and some things are specified in one build system, other things in the second build system.